CedarJava/src/main/java/com/cedarpolicy/model/policy/Policy.java

```
@@ -78,6 +78,24 @@ public String toString() {
         return "// Policy ID: " + policyID + "\n" + policySrc;
     }
 
+    /**
+     * Returns the effect of a policy. 
+     * 
+     * Determines the policy effect by attempting static policy first, then template. 
+     * In future, it will only support static policies once new class is introduced for Template.
+     * 
+     * @return The effect of the policy, either "permit" or "forbid"
+     * @throws InternalException
+     * @throws NullPointerException
+     */
+    public String effect() throws InternalException, NullPointerException {
+        try {
+            return policyEffectJni(policySrc); // Get effect for static policy
```

One thought I had here: does the order in which we try here matter?

I.e., we'd never see the exception thrown from `policyEffectJni`, only that from `templateEffectJni`. The policy exception would be swallowed. Is that fine?

As it stands now, I think the exceptions are similar and not specific to either static policies or templates, so we should be fine.

But say that I thought I had a `Policy` Java object for my static policy, and there was another issue with the policy with regards to this method. I would never see the relevant exception because we'd see an exception from failing to get the effect as a template.

Since we have to throw one of the exceptions anyway, there's not much we can do about it I don't think. It's an unfortunate symptom of the current "single class representing both Policy and Template" issue.

---

IIRC, it should be sufficient to only call the `templateEffectJni` without bothering to try parsing as a policy first. This isn't any better for the perspective a caller who thinks they have a policy rather than a template, but it would avoid needing to swallow the exception here

---

> I agree. One way to solve it could be to create a new exception class specifically for "Policy type and method mismatch." But as we plan to separate Policy and Template, I wonder if it would be worthwhile to create the new exception as it will probably not be thrown once the classes are separated.

Yeah agreed I don't think it warrants it either.